File: prototype/app/src/main/java/pt/isel/unicommunityprototype/model/Board.kt
package pt.isel.unicommunityprototype.model

data class Board(
    val id: Int,
    val name: String,
    val description: String?,
    private val modules: List<Module>, //TODO: forum is a module, but we want to distinguish forum from blackboards for easy access to forum!! Find a better way of modeling these classes
    val forum: Forum?
) {
    fun getAnunciosBlackboard() : Anuncios? {
        for (module in modules) {
            if (module is Anuncios) return module
        }

        return null
    }
}

File: client/src/components/board_details/model/Board.js
'use strict'
/**
 * @param {string} name        - The name of the board
 * @param {string} description - The description text of the board
 * @param {Object} modules     - A container object for the modules of the board. The object has the format:
 * {
 *     blackboards: Blackboard[],
 *     forum: array
 * }
 * Blackboard:
 * {
 *     name: string,
 *     content: array<Item>
 * }
 * 
 * Item has the format:
 * {
 *     name: string,
 *     text: string
 * }
 */
export default function Board(name, description, modules) {
    this.name = name
    this.description = description
    this.modules = modules
}